zenith_utils/src/pq_proto.rs

```diff
@@ -100,6 +100,21 @@ pub struct FeExecuteMessage {
 #[derive(Debug)]
 pub struct FeCloseMessage {}
 
+/// Retry a read on EINTR
+///
+/// This runs the enclosed expression, and if it returns
+/// Err(io::ErrorKind::Interrupted), retries it.
+macro_rules! retry_read {
```

Comment:
Why it is necessary to define `retry_read` as macro, not as a function?

Reply:
I tried that first. Maybe it'd be possible, but I couldn't make it work.